salt/states/github.py

```diff
@@ -583,26 +598,134 @@ def repo_present(
+                if __opts__['test']:
+                    test_comments.append('Repo {0} is set to be removed from team {1}.'
+                                         .format(name, team_name))
+                    ret['result'] = None
+                else:
+                    result = __salt__['github.remove_team_repo'](name, team_name,
+                                                                 profile=profile)
+                    if result:
+                        ret['changes'][team_name] = {
+                            'old': 'Repo {0} is in team {1}'.format(name, team_name),
+                            'new': 'Repo {0} is not in team {1}'.format(name, team_name)
+                        }
+                    else:
+                        ret['result'] = False
+                        ret['comment'] = ('Failed to remove repo {0} from team {1}.'
+                                          .format(name, team_name))
+                        return ret
+
+        # Next add or modify any necessary teams
+        for team_name, permission in six.iteritems(teams):
+            if team_name not in current_team_names:  # Need to add repo to team
+                if __opts__['test']:
+                    test_comments.append('Repo {0} is set to be added to team {1}.'
+                                         .format(name, team_name))
+                    ret['result'] = None
+                else:
+                    result = __salt__['github.add_team_repo'](name, team_name,
+                                                              permission,
+                                                              profile=profile)
+                    if result:
+                        ret['changes'][team_name] = {
+                            'old': 'Repo {0} is not in team {1}'.format(name, team_name),
+                            'new': 'Repo {0} is in team {1}'.format(name, team_name)
+                        }
+                    else:
+                        ret['result'] = False
+                        ret['comment'] = ('Failed to remove repo {0} from team {1}.'
+                                          .format(name, team_name))
+                        return ret
+            else:
+                current_permission = (__salt__['github.list_team_repos']
+                                      (team_name, profile=profile)
+                                      .get(name.lower(), {})
+                                      .get('permission'))
+                if not current_permission:
+                    ret['result'] = False
+                    ret['comment'] = ('Failed to determine current permission for team '
+                                      '{0} in repo {1}'.format(team_name, name))
+                    return ret
+                elif current_permission != permission:
+                    if __opts__['test']:
+                        test_comments.append('Repo {0} is set to have permissions changed '
```

Comment:
Same comment here as well. If changes will be made (which is what we're indicating by setting `ret['result'] = None`) then we should be able to see what is expected to change in the old/new option.